### PR TITLE
Updated doc gen for proposed new site design

### PR DESCRIPTION
**Pull request by parlough**
Depended on by: https://github.com/dart-lang/linter/pull/2115

Read the linked PR for more information, but this implements some minor cleanup and organizational changes for the proposed new site design. 

As I mentioned on the other PR, please feel free to to present heavy feedback and large changes.

### CODE REVIEW

File: tool/doc.dart [hunk 1]
```diff
@@ -268,7 +271,8 @@ class Generator {
       sb.writeln('<p>');
       sb.write('Incompatible with: ');
       var rule = incompatibleRules.first;
-      sb.write('<a href = "$rule.html" >$rule</a>');
+      sb.write(
+          '<!--suppress HtmlUnknownTarget --><a href = "$rule.html" >$rule</a>');
```
Comment (pq) [inline tool/doc.dart:275]: 💎

**Comment by parlough** [comment]
@pq You make a good point. When I was making these changes, it was more from a consistency standpoint to include the style button on more pages.

Perhaps we can remove this from the top bar and include it in a similar place to other guidelines. Then the top bar buttons could be reserved solely for the list of rules and if desired something else like a usage guide page or something else you have in mind.

**Comment by pq** [comment]
> Perhaps we can remove this from the top bar and include it in a similar place to other guidelines.

SGTM.  Thanks!

On a related note, it occurs to me that we had issues with navigation on mobile.  (See https://github.com/dart-lang/linter/issues/1324.)  I wonder: does  this arc of work fix that?  (Or could you take a look perhaps while you're at it?)

Thanks a million for taking all of this on!

**Comment by parlough** [comment]
@pq I went ahead and kept information about the style good near other rule sets and opted to include a link to more information about using the Linter.

As for being more mobile friendly, the buttons fit on a lot of screen sizes, but when they do not you get something like this:

**Comment by pq** [comment]
Beautiful.  Are we ready to land?

Thanks so much!

**Comment by parlough** [comment]
> Beautiful.  Are we ready to land?
> 
> Thanks so much!

If you're happy with all of the changes, the PRs are good from me :). Can always make further changes and cleanup later.

**Comment by pq** [comment]
Excellent.  I'll merge and push a new version of the docs.  Thanks!

Could you give me attribution info for https://github.com/dart-lang/linter/blob/master/AUTHORS?

**Pull request merged**